ADC_DMA/Core/Src/main.c

```diff
@@ -0,0 +1,184 @@
+
+int main(void)
+{
+	sADC_measure 	ADC_measure = {0};	//struct with all the information read by the ADC channel
+	uint16_t 		aux = 0;			//we only use aux to calculate v_ref
+
+	/*							  *
+	 * --- DMA INITIALIZATION --- *
+	 *							  */
+	RCC->AHBENR |= RCC_AHBENR_DMAEN;
+	DMA1_Channel1->CPAR	= 	(uint32_t) &(ADC1->DR);		//setting peripheral address
+	DMA1_Channel1->CCR 	|=	DMA_CCR_CIRC	|			//DMA in circular mode
+							DMA_CCR_MINC	|			//memory increment
+							DMA_CCR_PSIZE_0	|			//peripheral size set to 16 bits
+							DMA_CCR_MSIZE_0	;			//memory size set to 16 bits
+
+	/*										*
+	 * --- ADC INITIALIZATION PROCEDURE --- *
+	 *										*/
+	RCC->APB2ENR 	|= 	RCC_APB2ENR_ADC1EN;		//enables ADC clock
+	ADC1->CR 		|=	ADC_CR_ADVREGEN;		//enables the voltage regulator
+	ADC1->CFGR2 	|= 	ADC_CFGR2_CKMODE_0;		//sets ADC clock as PCLK/2
+	ADC->CCR 		|= 	ADC_CCR_LFMEN;			//low frequency mode enable
+	ADC1->SMPR 		|= 	ADC_SMPR_SMPR_2;		//sets sample time for 19.5 ADC clock cycles (14 needed)
+	ADC1->CFGR1		|=	ADC_CFGR1_WAIT;			//enables wait mode to prevent overrun
+
+	ADC1->CR |= ADC_CR_ADCAL;					//starting the calibration
+	while(ADC1->CR & ADC_CR_ADCAL);				//we have to wait until ADCAL = 0 (Can be handled by interrupt)
+
+	ADC1->CFGR1	|=	ADC_CFGR1_DMACFG |			//DMA in circular mode
+					ADC_CFGR1_DMAEN	;			//enables DMA	- must be done AFTER calibrate the AD
+
+	ADC1->ISR|= ADC_ISR_ADRDY;					//clear the ADRDY bit by programming it to 1
+	ADC1->CR |= ADC_CR_ADEN;					//then we enable the ADC
+	while(!(ADC1->ISR & ADC_ISR_ADRDY));		//and wait for it to be ready (Can be handled by interrupt)
+
+	/*												*
+	 * --- ADC CHANNEL CONFIG FOR V_REF READING --- *
+	 *												*/
+	if(ADC1->CR & ADC_CR_ADSTART)					//we have to be sure there's no ongoing conversion
+	{
+		ADC1->CR |= ADC_CR_ADSTP;					//if so, we have to set ADSTP bit
+		while(ADC1->CR & ADC_CR_ADSTP);				//and wait for it to clear
+	}
+	ADC1->CR |= ADC_CR_ADDIS;						//disables the ADC
+	while(ADC1->CR & ADC_CR_ADEN);
+
+	ADC1->CHSELR = 	ADC_CHSELR_CHSEL17	|			//selecting the VREF channel
+					ADC_CHSELR_CHSEL18	;			//and Tsense channel
+	ADC->CCR 	|= 	ADC_CCR_VREFEN	| 				//enables internal reference voltage
+					ADC_CCR_TSEN	; 				//and temperature sensor
+	wait(TIME_10uSEC);								//we have to wait for the proper time for the Tsense to wake up
+
+	DMA1_Channel1->CNDTR = 2; 								//number of data to be transferred
+	DMA1_Channel1->CMAR = (uint32_t) &(ADC_measure.v_ref);	//set the first memory register address for DMA
+	DMA1_Channel1->CCR 	|= DMA_CCR_EN;						//enables the DMA
+
+	ADC1->CR |= ADC_CR_ADEN;						//then we enable the ADC
+	while(!(ADC1->ISR & ADC_ISR_ADRDY));			//and wait for it to be ready. (Can be handled by interrupt)
+
+	ADC1->CR |= ADC_CR_ADSTART;						//starts the ADC
+	while(1)
+	{
+		//we wait for the ADC and DMA to complete the process
+		if((ADC1->ISR & ADC_ISR_EOS) && (DMA1->ISR & DMA_ISR_TCIF1))
```

Comment:
Why are we checking EOS and TCIF? Won't it mean basically the same thing?

According to the RM, EOS indicates the end of sequence conversion and TCIF indicates that DMA has transferred all the data to its destination. I understand that what happens is TCIF goes high right after EOS, meaning that we only need to check TCIF flag to be sure that our data is ready.